python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -541,18 +538,20 @@ class RandomForestClassifier(Base):
 
     def _predict_model_on_gpu(self, X, output_class,
                               threshold, algo, num_classes):
-        _, _, n_rows, n_cols, _ = \
+        _, _, n_rows, n_cols, X_dtype = \
             input_to_dev_array(X, order='C')
         if n_cols != self.n_cols:
             raise ValueError("The number of columns/features in the training"
                              " and test data should be the same ")
+        if X_dtype != self.dtype:
```

Comment:
you should do this check as part of the `input_to_dev_array` call instead of here